test/server_test.go

```diff
@@ -68,3 +72,37 @@ func TestClientSideError(t *testing.T) {
 	_, ok := err.(models.DefaultInternalError)
 	assert.True(t, ok)
 }
+
+type MiddlewareContextTest struct {
```

Comment:
at some point we should probably use gomock, or something like that for this. if nothing else, for an example that others can copy